statsmodels/regression/process_reg.py

```diff
@@ -0,0 +1,724 @@
+        params : array-like
+            The packed parameters for the model.
+
+        Returns
+        -------
+        The score vector at the given parameter point.
+
+        Notes
+        -----
+        The mean, scaling, and smoothing parameters are packed into
+        a vector.  Use `unpack` to access the component vectors.
+        """
+
+        mnpar, sdpar, smpar = self.unpack(params)
+        pm, pv = len(mnpar), len(sdpar)
+
+        # Residuals
+        resid = self.endog - np.dot(self.exog, mnpar)
+
+        # Standard deviations
+        sd = np.exp(np.dot(self.exog_scale, sdpar))
+
+        # Smoothness
+        sm = np.exp(np.dot(self.exog_smooth, smpar))
+
+        # Get the log-likelihood
+        score = np.zeros(len(mnpar) + len(sdpar) + len(smpar))
+        for _, ix in self._groups_ix.items():
+
+            sd_i = sd[ix]
+            sm_i = sm[ix]
+            resid_i = resid[ix]
+            time_i = self.time[ix]
+            exog_i = self.exog[ix, :]
+            exog_scale_i = self.exog_scale[ix, :]
+            exog_smooth_i = self.exog_smooth[ix, :]
+
+            # Get the covariance matrix for this person.
+            cm = self.cov.get_cov(time_i, sd_i, sm_i)
+            cmi = np.linalg.inv(cm)
+
+            jacv, jacs = self.cov.jac(time_i, sd_i, sm_i)
+
+            # The derivatives for the mean parameters.
+            dcr = np.linalg.solve(cm, resid_i)
+            score[0:pm] += np.dot(exog_i.T, dcr)
+
+            # The derivatives for the standard deviation parameters.
+            rx = np.outer(resid_i, resid_i)
+            qm = np.linalg.solve(cm, rx)
+            qm = 0.5 * np.linalg.solve(cm, qm.T)
+            sdx = sd_i[:, None] * exog_scale_i
+            for i, _ in enumerate(ix):
+                jq = np.sum(jacv[i] * qm)
+                score[pm:pm + pv] += jq * sdx[i, :]
+                score[pm:pm + pv] -= 0.5 * np.sum(jacv[i] * cmi) * sdx[i, :]
+
+            # The derivatives for the smoothness parameters.
+            smx = sm_i[:, None] * exog_smooth_i
+            for i, _ in enumerate(ix):
+                jq = np.sum(jacs[i] * qm)
+                score[pm + pv:] += jq * smx[i, :]
+                score[pm + pv:] -= 0.5 * np.sum(jacs[i] * cmi) * smx[i, :]
+
+        if self.verbose:
+            print("|G|=", np.sqrt(np.sum(score * score)))
+
+        return score
+
+    def fit(self, start_params=None, method=None, maxiter=None,
+            full_output=True, disp=True, fargs=(), callback=None,
+            retall=False, skip_hessian=False, **kwargs):
+
+        if "verbose" in kwargs:
+            self.verbose = kwargs["verbose"]
+
+        minim_opts = {}
+        if "minim_opts" in kwargs:
+            minim_opts = kwargs["minim_opts"]
+
+        if start_params is None:
+            start_params = self._get_start()
+
+        if isinstance(method, str):
+            method = [method]
+        elif method is None:
+            method = ["powell", "bfgs"]
```

Comment:
We have an old PR by Skipper where he added support for using a sequence of optimizers.
We could revisit that if that has sufficient use cases.

Reply:
The fit method looks standard and could use super.fit instead, except for the sequence of methods